src/SystemCtl.php

```diff
@@ -70,105 +74,121 @@ public static function unitFromSuffix(string $unitSuffix, string $unitName): Uni
             throw new UnitTypeNotSupportedException('Unit type ' . $unitSuffix . ' not supported');
         }
 
-        return new $unitClass($unitName, new ProcessBuilder([self::$binary]));
+        $commandDispatcher = (new SymfonyCommandDispatcher)
+            ->setTimeout(self::$timeout)
+            ->setBinary(self::$binary);
+
+        return new $unitClass($unitName, $commandDispatcher);
     }
 
     /**
      * List all supported units
      *
      * @param null|string $unitPrefix
-     * @param string[] $unitTypes
+     * @param string[]    $unitTypes
+     *
      * @return array|\string[]
      */
     public function listUnits(?string $unitPrefix = null, array $unitTypes = self::SUPPORTED_UNITS): array
     {
-        $processBuilder = $this->getProcessBuilder()
-            ->add('list-units');
+        $commands = ['list-units'];
```

Comment:
> The problem with the add syntax is that this was because of the hard depedency of Symfony\ProcessBuilder since this is gone.

It is not gone, it is just abstracted. In the context of the abstract `Dispatcher`, it is true, that does not make any big sense.

I would have stuck to the terminology you had before more and created an abstract CommandBuilder, that would have lean on the way Symfony build all that.